### PR TITLE
Add Korean language (UTF-32) support

### PR DESCRIPTION
**Pull request by ycs1m1yk**
Implemented UTF-32 class based on existing UTF-16 class.

Currently, only BMP(U+0000 - U+FFFF) is supported.

All test-cases from #95 are passed!

Feel free to let me know if there are any errors or anything that needs to be improved. 😎

### CODE REVIEW

**Comment by kuro11pow2** [comment]
Thank you for your contribution! 👍

**Comment by aiclaudev** [comment]
Great!! Thanks 👍

**Comment by jopopcorn** [comment]
Good job! It's a great contribution!

File: euphony/src/main/cpp/CMakeLists.txt [hunk 1]
```diff
@@ -27,6 +27,7 @@ set(EUPHONY_SRC
         core/source/AudioStreamCallback.cpp
         core/source/ASCIICharset.cpp
         core/source/UTF16Charset.cpp
+        core/source/UTF32Charset.cpp
```
Comment (designe) [inline euphony/src/main/cpp/CMakeLists.txt:30]: Nicely done with adding source in CmakeLists!

File: euphony/src/main/cpp/core/source/UTF32Charset.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    for (int hexIdx = 0; hexIdx < hexSource.size(); hexIdx+=HEX_NUM_COUNT) {
+        char32_t ch = hexSource[hexIdx];
+        ch = ( ch << BIT_COUNT_IN_HEX_NUM ) | hexSource[hexIdx + 1];
+        ch = ( ch << BIT_COUNT_IN_HEX_NUM ) | hexSource[hexIdx + 2];
+        ch = ( ch << BIT_COUNT_IN_HEX_NUM ) | hexSource[hexIdx + 3];
+        ch = ( ch << BIT_COUNT_IN_HEX_NUM ) | hexSource[hexIdx + 4];
+        ch = ( ch << BIT_COUNT_IN_HEX_NUM ) | hexSource[hexIdx + 5];
+        ch = ( ch << BIT_COUNT_IN_HEX_NUM ) | hexSource[hexIdx + 6];
+        ch = ( ch << BIT_COUNT_IN_HEX_NUM ) | hexSource[hexIdx + 7];
```
Comment (designe) [inline euphony/src/main/cpp/core/source/UTF32Charset.cpp:41]: I don't like multiple iteration :(
but, it seems that these 7 lines are necessary. how do you think about it?

File: euphony/src/main/cpp/tests/utf32CharsetTest.cpp [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                TestParamType("kācaṃ śaknomyattum; nopahinasti mām", "0000006b00000101000000630000006100001e43000000200000015b000000610000006b0000006e0000006f0000006d00000079000000610000007400000074000000750000006d0000003b000000200000006e0000006f000000700000006100000068000000690000006e00000061000000730000007400000069000000200000006d000001010000006d"),
+                TestParamType("សប្បាយ", "0000179f00001794000017d200001794000017b600001799"),
+                TestParamType("᚛᚛ᚉᚑᚅᚔᚉᚉᚔᚋ ᚔᚈᚔ ᚍᚂᚐᚅᚑ ᚅᚔᚋᚌᚓᚅᚐ᚜", "0000169b0000169b000016890000169100001685000016940000168900001689000016940000168b00001680000016940000168800001694000016800000168d000016820000169000001685000016910000168000001685000016940000168b0000168c0000169300001685000016900000169c"),
+                TestParamType("⠊⠀⠉⠁⠝⠀⠑⠁⠞⠀⠛⠇⠁⠎⠎⠀⠁⠝⠙⠀⠊⠞⠀⠙⠕⠑⠎⠝⠞⠀⠓⠥⠗⠞⠀⠍⠑", "0000280a0000280000002809000028010000281d0000280000002811000028010000281e000028000000281b00002807000028010000280e0000280e00002800000028010000281d00002819000028000000280a0000281e000028000000281900002815000028110000280e0000281d0000281e000028000000281300002825000028170000281e000028000000280d00002811")
```
Comment (designe) [inline euphony/src/main/cpp/tests/utf32CharsetTest.cpp:82]: fantastic unit-test cases :)

**Review by designe — COMMENTED** [review]
Thanks for your great utf-32 support contribution.
I leave some comments!

**Comment by eunzinn** [comment]
Gooood!!!! 👍 👍 👍

**Comment by judemin** [comment]
Great Job!

File: euphony/src/main/cpp/core/source/UTF32Charset.cpp (hunk 2 above)
Comment (designe) [reply euphony/src/main/cpp/core/source/UTF32Charset.cpp]: Great :)

**Pull request merged**